bootstrap.sh

```diff
@@ -1,39 +1,505 @@
 #!/bin/bash
+#
```

Comment:
if you use sudo in a long script, request it early and make sure it doesn't timeout until something down the line needs it